Guides/UPI/okd4-terraform-openstack/README.md

```diff
@@ -0,0 +1,120 @@
+download directly from the realease pages of OKD and Fedora CoreOS and manually
+do what the script does with any combination of OKD build and Fedora CoreOS
+build. Just be aware that the image dependencies used by an nightly build may or
+may not be available on quay.io.
+
+Ref:
+
+* https://builds.coreos.fedoraproject.org/browser
+* https://origin-release.svc.ci.openshift.org/
+
+## Configuring the deployment
+
+First edit the `terraform.tfvars` file and fill in according to the comments.
+
+PS: The load balancer uses a standard Centos 7 image that must be present in
+the Openstack project. The default name is found in the `variables.tf` but can
+be overridden in terraform.tfvars (as all other default variable values) to
+what the name of the image is in your project.   
+
+Then edit the install-config.yaml and put in the parameters according to comments.
+
+## Deploying the cluster
+
+In addition to the previous requirements you also need access to a s3 bucket in
+order to store the ignition config for the bootstrap node which is so huge that it
+will grow the user_data field inside the POST request to openstack so much that
+creating the boot instance fails due to exeeded size limit in the Openstack
+API.
+
+The `deploy.sh` script takes two arguments: `install-config.yaml` and
+`s3-endpoint-url` of the s3 object store to first put and subsequently fetch
+the ignition config of the bootstrap node.
+
+You must have the `aws` cli installed and configured with credentials and
+privileges to to put and presign the ignition file to the s3-ignition-url.
+
+When it is in place you can run the deploy script:
+
+`scripts/deploy.sh`
+
+The first thing that happens is that it creates manifests that unfortunately
+have to be modified before further processing. So the script stops and asks you
+to edit `installer/manifests/cluster-scheduler-02-config.yml` (which was just
+created). Change `mastersSchedulable: true` to `mastersSchedulable: False`. Then
+hit return to continue.
+
+The script will frist run terraform with paramaters that override default
+values in `terraform.tfvars` in order to start out only with 3 control plane
+nodes and one bootstrap node. When bootstrap is complete terraform desired
+state is applied with default values (which means 0 boot nodes and N worker
+nodes. Loadbalancer (haproxy) config are applied and updated several times  in
+between in order to reflect the changes in the set of nodes during bootstrap.
+
+During the bootstrap you can follow the progress of cluster operators with:
+
+```
+export KUBECONFIG=installer/auth/kubeconfig
+watch ./oc get co
+```
+
+Finally csr's of worker nodes are approved, (or actually all pending csr's are
+approved, assuming it is the nodes that requested them, Be aware)
+
+When all cluster operators are ready, you should be able to log in to the web
+console. To find the address:
+
+` ./oc get route --all-namespaces |grep ' console'`
```

Comment:
Use `wait-for install-complete` in the installer? This should output console URL and kubeadmin password afaik

Reply:
Thanks, I'll change it.